Components/Graph/TestSuite/GraphTests.cpp

```diff
@@ -0,0 +1,659 @@
+
+  // View Types
+  std::array<std::array<std::string, 2>, 3> layOpt;
+  layOpt[0] = {"DCCSC3D", "DCCSC3D"};
+  // layOpt[1] = {"DCCSC3D", "S1CLCSC3D"};
+  // layOpt[2] = {"DCCSC3D", "DCCSC3D"};
+
+  auto mem = std::make_shared<QuICC::Memory::Cpu::NewDelete>();
+  using namespace QuICC::Graph;
+  std::vector<QuICC::View::ViewBase<std::uint32_t>> meta;
+  Jit<rank> Jitter(modStr, mem, physDims, modsDims, layOpt, Stage::MPP, Stage::MPP, meta);
+
+  // setup metadata
+  auto modsM = modsDims[2];
+  auto N = physDims[1];
+  auto K = physDims[0];
+  std::array<std::uint32_t, 3> modsDimensions {modsM, N, K};
+
+  std::array<std::vector<std::uint32_t>, 3> pointers {{{},{0, 2, 2, 2, 3, 4},{}}};
+  std::array<std::vector<std::uint32_t>, 3> indices {{{},{0, 1, 0, 0},{}}};
+
+  // host mem block
+  std::size_t modsS = modsM*indices[1].size();
+  QuICC::Memory::MemBlock<std::complex<double>> modsIn(modsS, mem.get());
+  QuICC::Memory::MemBlock<std::complex<double>> modsOut(modsS, mem.get());
+
+  // host view
+  using namespace QuICC::Graph;
+  C_DCCSC3D_t modsInView({modsIn.data(), modsIn.size()}, modsDimensions, pointers, indices);
+  C_DCCSC3D_t modsOutView({modsOut.data(), modsOut.size()}, modsDimensions, pointers, indices);
+
+  // set input modes
+  std::complex<double> val = {1.0, 0.0};
+  for(std::size_t m = 0; m < modsInView.size(); ++m)
+  {
+    modsInView[m] = val;
+  }
+
+  // Apply graph
+  Jitter.apply(modsOutView, modsInView);
+
+  // Check
+  for(std::size_t m = 0; m < modsOutView.size(); ++m)
+  {
+    CHECK(modsOutView[m] == val);
+  }
+}
+
+TEST_CASE("One Dimensional Loop Associated Legendre", "[OneDimLoopAL]")
+{
+  // Test Graph
+  // Same setup as transform loop
+  std::string modStr = R"mlir(
+    func.func @entry(%tumod: tensor<?x?x?xcomplex<f64>>) -> (tensor<?x?x?xcomplex<f64>>) {
+      %tuval = quiccir.al.prj %tumod : tensor<?x?x?xcomplex<f64>> -> tensor<?x?x?xcomplex<f64>> attributes{kind = "P"}
+      %ret = quiccir.al.int %tuval : tensor<?x?x?xcomplex<f64>> -> tensor<?x?x?xcomplex<f64>> attributes{kind = "P"}
+      return %ret : tensor<?x?x?xcomplex<f64>>
+    }
+  )mlir";
+
+  // Grid dimensions
+  constexpr std::uint32_t rank = 3u;
+  // v012
+  std::array<std::uint32_t, rank> physDims{1, 20, 4};
+  std::array<std::uint32_t, rank> modsDims{1, 10, 4};
+
+  // View Types
+  std::array<std::array<std::string, 2>, 3> layOpt;
+  // layOpt[0] = {"DCCSC3D", "DCCSC3D"};
+  layOpt[1] = {"DCCSC3D", "S1CLCSC3D"};
+  // layOpt[2] = {"DCCSC3D", "DCCSC3D"};
+
+  auto mem = std::make_shared<QuICC::Memory::Cpu::NewDelete>();
+  using namespace QuICC::Graph;
+  std::vector<QuICC::View::ViewBase<std::uint32_t>> meta;
+  Jit<rank> Jitter(modStr, mem, physDims, modsDims, layOpt, Stage::MPM, Stage::MPM, meta);
+
+  // setup metadata
+  auto M = physDims[2];
+  // auto N = physDims[1];
+  auto K = physDims[0];
+  auto modsM = modsDims[2];
+  auto modsN = modsDims[1];
+  // auto modsK = modsDims[2];
```

Comment:
Needed? and looks wrong

Reply:
should be zero, fixed, also here is for doumentation